allennlp/training/trainer.py

```diff
@@ -758,8 +759,6 @@ def train(self) -> Dict[str, Any]:
 
                     # Check validation metric to see if it's the best so far
                     is_best_so_far = self._is_best_so_far(this_epoch_val_metric, validation_metric_per_epoch)
-                    if is_best_so_far:
-                        best_epoch_val_metrics = val_metrics.copy()
```

Comment:
This is still necessary, we don't want to only have the best metrics present if _this epoch_ is the best